src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -358,10 +358,48 @@ private async Task SetPackagePropertyValueAsync(IProjectProperties metadata, str
                 propertyValue);
         }
 
-        public override async Task<bool> UninstallPackageAsync(PackageIdentity packageIdentity, INuGetProjectContext nuGetProjectContext, CancellationToken token)
+        public override Task<bool> UninstallPackageAsync(PackageIdentity packageIdentity, INuGetProjectContext nuGetProjectContext, CancellationToken token)
         {
-            var configuredProject = await _unconfiguredProject.GetSuggestedConfiguredProjectAsync();
-            await configuredProject?.Services.PackageReferences.RemoveAsync(packageIdentity.Id);
+            throw new InvalidOperationException("For CPSPackageReferenceProject, the uninstall method that can handle conditional targeting must be called instead.");
+        }
+
+        public override async Task<bool> UninstallPackageAsync(string packageId, BuildIntegratedInstallationContext installationContext, CancellationToken token)
+        {
+            token.ThrowIfCancellationRequested();
+            if (string.IsNullOrEmpty(packageId)) throw new ArgumentException(string.Format(Strings.Argument_Cannot_Be_Null_Or_Empty, nameof(packageId)));
+            if (installationContext == null) throw new ArgumentNullException(nameof(installationContext));
+
+            if (installationContext.SuccessfulFrameworks.Any() && installationContext.UnsuccessfulFrameworks.Any())
```

Comment:
I am curious why are we checking `installationContext.UnsuccessfulFrameworks.Any()` because it is not used in the if block.

Reply:
We only want to do conditional uninstallation if there are both successful and unsuccessful frameworks.